src/pages/Intents/CustomIntentsList.js

```diff
@@ -0,0 +1,52 @@
+import { CircularLoader } from '@dhis2/ui'
+import isEqual from 'lodash/isEqual'
+import PropTypes from 'prop-types'
+import React, { useEffect, useMemo } from 'react'
+import TableActions from '../../components/TableActions'
+import NewIntent from './NewIntent'
+
+// TODO: create function to model the data that will be save in Datastore, get Data elements and attributes
+const CustomIntentsList = ({ settings, handleSettings, disable }) => {
+    const loading = false
+    const rows = settings
+    const initialRows = useMemo(() => {
+        return settings
+    }, [settings])
+
+    useEffect(() => {
+        if (rows && initialRows && !isEqual(rows, initialRows)) {
+            handleSettings(rows)
+        }
+    }, [rows])
+
+    return (
+        <>
+            {loading && <CircularLoader />}
+            {rows.length > 0 && <RowList rows={rows} disable={disable} />}
+
+            <NewIntent disable={disable} />
```

Comment:
I think it would probably be easier to put this at CustomIntents.js level, so then this CustomIntentsList remains more strictly just a UI rendering of the current list?